catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,706 @@
+            serviceConfigurationProperties.put(ID, configId);
+            serviceConfigurationProperties.put(SHORTNAME, configId);
+            serviceConfigurationProperties.put(RegistryObjectMetacardType.REGISTRY_ID, registryId);
+
+            curConfig.update(serviceConfigurationProperties);
+        }
+
+    }
+
+    /**
+     * Finds a configuration in the map of current configurations or creates a new one if one doesn't
+     * exist yet.
+     *
+     * @param factoryPid             The factory pid to lookup/create
+     * @param fpidToConfigurationMap Map containing current configurations
+     * @param createActiveConfig     If a configuration needs to be created, indicates if it should
+     *                               be created as an active or disable configuration.
+     * @return
+     * @throws IOException
+     */
+    private Configuration findOrCreateConfig(String factoryPid,
+            Map<String, Configuration> fpidToConfigurationMap, boolean createActiveConfig)
+            throws IOException {
+        String factoryPidDisabled = factoryPid.concat(DISABLED_CONFIGURATION_SUFFIX);
+
+        Configuration curConfig = fpidToConfigurationMap.get(factoryPid);
+        if (curConfig == null) {
+            curConfig = fpidToConfigurationMap.get(factoryPidDisabled);
+        }
+        if (curConfig == null) {
+            String pid = factoryPidDisabled;
+            if (createActiveConfig) {
+                pid = factoryPid;
+            }
+            curConfig = configurationAdmin.createFactoryConfiguration(pid, null);
+        }
+        return curConfig;
+    }
+
+    /**
+     * Activates an existing configuration if it isn't active and has the fpid matching the fPidToActivate.
+     * If there is an acticve configuration that isn't the fPidToActivate it will deactivate it.
+     *
+     * @param fPidToActivate         Factory PID that should be active
+     * @param fpidToConfigurationMap Map of existing fpid -> configuration mappings
+     * @throws IOException
+     */
+    private void activateDeactivateExistingConfiguration(String fPidToActivate,
+            Map<String, Configuration> fpidToConfigurationMap) throws IOException {
+        String fpid;
+        Configuration configToEnable = null;
+        Configuration configToDisable = null;
+        for (Map.Entry<String, Configuration> entry : fpidToConfigurationMap.entrySet()) {
+            fpid = entry.getKey();
+            if (!fpid.contains(DISABLED_CONFIGURATION_SUFFIX) && !fpid.equals(fPidToActivate)) {
+                configToDisable = entry.getValue();
+            } else if (fpid.equals(fPidToActivate.concat(DISABLED_CONFIGURATION_SUFFIX))) {
+                configToEnable = entry.getValue();
+            }
+        }
+        //Order of disable/enable important. Can't have two enabled configurations with the same
+        //id so if there is one to disable, disable it before enabling another one.
+        if (configToDisable != null) {
```

Comment:
So I just realized that only one of those will be returned by the loop above so the stream solution doesn't really make sense.

Would it be beneficial however to exist the for loop when we know both `configToDisable` and `configToEnable` have been found?